storage/aptosdb/src/state_restore/mod.rs

```diff
@@ -191,45 +245,63 @@ impl<K: Key + CryptoHash + Hash + Eq, V: Value> StateSnapshotReceiver<K, V>
     for StateSnapshotRestore<K, V>
 {
     fn add_chunk(&mut self, chunk: Vec<(K, V)>, proof: SparseMerkleRangeProof) -> Result<()> {
-        let _timer = OTHER_TIMERS_SECONDS
-            .with_label_values(&["state_snapshot_add_chunk"])
-            .start_timer();
+        let kv_fn = || {
+            let _timer = OTHER_TIMERS_SECONDS
+                .with_label_values(&["state_value_add_chunk"])
+                .start_timer();
+            self.kv_restore
+                .lock()
+                .as_mut()
+                .unwrap()
+                .add_chunk(chunk.clone())
+        };
+
+        let tree_fn = || {
+            let _timer = OTHER_TIMERS_SECONDS
+                .with_label_values(&["jmt_add_chunk"])
+                .start_timer();
+            self.tree_restore
+                .lock()
+                .as_mut()
+                .unwrap()
+                .add_chunk_impl(chunk.iter().map(|(k, v)| (k, v.hash())).collect(), proof)
+        };
         // Write KV out first because we are likely to resume according to the rightmost key in the
         // tree after crashing.
-        let (r1, r2) = IO_POOL.join(
-            || {
-                let _timer = OTHER_TIMERS_SECONDS
-                    .with_label_values(&["state_value_add_chunk"])
-                    .start_timer();
-                self.kv_restore
-                    .lock()
-                    .as_mut()
-                    .unwrap()
-                    .add_chunk(chunk.clone())
-            },
-            || {
-                let _timer = OTHER_TIMERS_SECONDS
-                    .with_label_values(&["jmt_add_chunk"])
-                    .start_timer();
-                self.tree_restore
-                    .lock()
-                    .as_mut()
-                    .unwrap()
-                    .add_chunk_impl(chunk.iter().map(|(k, v)| (k, v.hash())).collect(), proof)
+        match self.restore_mode {
+            StateSnapshotRestoreMode::KvOnly => kv_fn()?,
+            StateSnapshotRestoreMode::TreeOnly | StateSnapshotRestoreMode::Default => {
+                // We run kv_fn with TreeOnly to restore the usage of DB
```

Comment:
when in tree mode, the usage should've been done by previous operations already, right?

Reply:
no, I didn't update the usage when applying the writeset and only update the usage in batch in snapshot restore. Previously, we discussed it requires reading the current usage of the item corresponding to the WS if we want to update the usage while applying WS.